python/lsst/pipe/base/script/transfer_from_graph.py

```diff
@@ -62,13 +62,26 @@ def transfer_from_graph(
     qgraph = QuantumGraph.loadUri(graph)
 
     # Collect output refs that could be created by this graph.
-    output_refs: set[DatasetRef] = set(qgraph.globalInitOutputRefs())
+    original_output_refs: set[DatasetRef] = set(qgraph.globalInitOutputRefs())
     for task_def in qgraph.iterTaskGraph():
         if refs := qgraph.initOutputRefs(task_def):
-            output_refs.update(refs)
+            original_output_refs.update(refs)
     for qnode in qgraph:
         for refs in qnode.quantum.outputs.values():
-            output_refs.update(refs)
+            original_output_refs.update(refs)
+
+    # Get data repository definitions from the QuantumGraph; these can have
+    # different storage classes than those in the quanta.
+    dataset_types = {dstype.name: dstype for dstype in qgraph.registryDatasetTypes()}
+
+    # Convert output_refs to the data repository storage classes, too.
+    output_refs = set()
+    for ref in original_output_refs:
+        internal_dataset_type = dataset_types.get(ref.datasetType.name, ref.datasetType)
+        if internal_dataset_type.storageClass_name != ref.datasetType.storageClass_name:
+            output_refs.add(ref.overrideStorageClass(internal_dataset_type.storageClass_name))
```

Comment:
```suggestion
            output_refs.add(ref.overrideStorageClass(internal_dataset_type.storageClass))
```
since the storage class should be attached and can be used directly without having to proxy through the name string.

Reply:
If you think that's safe, sure.  I tend to default to using the name when I'm not sure because I don't want to accidentally trigger imports of things that might not be available, but that may just be a habit relevant for working down in the guts of registry, not here.

Reply:
I think at this point the storage class is going to be resolved into a real storage class almost immediately so it won't matter. I guess there's no harm in going via the name here and then deferring sotrage class creation as much as possible.